strings/base_events.h

```diff
@@ -375,8 +375,37 @@ WINRT_EXPORT namespace winrt
         using delegate_type = Delegate;
 
         event() = default;
-        event(event<Delegate> const&) = delete;
-        event<Delegate>& operator =(event<Delegate> const&) = delete;
+
+        event(event<Delegate> const& value)
+        {
+            slim_lock_guard const swap_guard(value.m_swap);
+            m_targets = value.m_targets;
+        }
+
+        event<Delegate>& operator =(event<Delegate> const& value)
```

Comment:
Also move assignment? ("No" is a valid answer.)

```cpp
event& operator=(event const& value)
{
    return assign(value);
}
event& operator=(event&& value)
{
    return assign(std::move(value));
}

private:
template<typename T>
event& assign(T&& value)
{
 ... existing code, except that
    slim_lock_guard const swap_guard(value.m_swap);
    new_targets = value.m_targets;

becomes
    if constexpr (std::is_lvalue_reference_v<T>)
    {
        slim_lock_guard const swap_guard(value.m_swap);
        new_targets = value.m_targets;
    }
    else
    {
        slim_lock_guard const swap_guard(value.m_change);
        slim_lock_guard const swap_guard(value.m_swap);
        new_targets = std::move(value.m_targets);
    }
...
}
```

Move assignment is tricky because
1. You have to worry about the move assignment throwing an exception midway. Not an issue here, since the delegate_array move assignment is nonthrowing.
2. You have a brief window in which both events are empty. I think that's okay. If you hit that window, then your code was already race-y and could validly have called nothing back.

If you are worried about code explosion, you can factor that constexpr into a helper.

```cpp
event& operator=(event const& value)
{
    return assign(value, [](auto&& v) {
        slim_lock_guard const swap_guard(v.m_swap);
        return v.m_targets;
    });
}
event& operator=(event&& value)
{
    return assign(value, [](auto&& v) {
        slim_lock_guard const swap_guard(v.m_change);
        slim_lock_guard const swap_guard(v.m_swap);
        return const_cast<event&&>(v).m_targets;
    });
}

event& assign(event const& value, delegate_array (*extract)(event const&))
{
... as before except
    new_targets = extract(value);
}
```

But I don't think code explosion is a problem because all of the event lvalue assignments will fold into one copy, and all of the event rvalue assignments will fold into another copy. So at most two copies.

Reply:
Move assignment/construction requires more locking than just copy assignment, so I believe a copy ends up being cheaper than a move.